In [1]:
import pandas as pd
import numpy as np
%matplotlib inline

In [2]:
df = pd.read_csv('pcos.csv')
df.head()

PCOS  weight_gain  hair_growth  skin_darkening  hair_loss
0   0.0          0.0          0.0             0.0        0.0
1   0.0          0.0          0.0             0.0        0.0
2   1.0          0.0          0.0             0.0        1.0
3   0.0          0.0          0.0             0.0        0.0
4   0.0          0.0          0.0             0.0        1.0

In [6]:
#prob of PCOS
df.PCOS.mean()

0.32717190388170053

In [7]:
#prob of skin darkening
df.skin_darkening.mean()

0.3068391866913124

In [9]:
pcos = df.query('PCOS == 1')
pcos

PCOS  weight_gain  hair_growth  skin_darkening  hair_loss
2     1.0          0.0          0.0             0.0        1.0
12    1.0          1.0          1.0             1.0        1.0
19    1.0          0.0          1.0             1.0        1.0
24    1.0          0.0          1.0             0.0        0.0
26    1.0          1.0          1.0             1.0        1.0
..    ...          ...          ...             ...        ...
520   1.0          0.0          1.0             0.0        0.0
523   1.0          1.0          0.0             0.0        0.0
524   1.0          1.0          0.0             0.0        0.0
533   1.0          0.0          0.0             0.0        0.0
540   1.0          1.0          1.0             1.0        1.0

[177 rows x 5 columns]

In [11]:
#avg no. with skin darkening (SD)
pcos.skin_darkening.mean()

0.6214689265536724

## Applying Bayes Theory
### Prior:
pcos = 0.327 
Nopcos = (1 - 0.327) = 0.673
SD-pcos = 0.621 
NoSD-pcos = (1 - 0.621) = 0.379
SD-Nopcos = (1 - 0.846) = 0.154
NoSD-Nopcos = 0.846

### Posterior:
Prob of having skin darkening and PCOS
P(PCOS|SD) = 0.327 * 0.621 = 0.203

Prob of having skin darkening without PCOS P(NoPCOS|SD)= 0.673 * 0.154 = 0.104

### Normalizer:
Prob of skin darkening = P(PCOS|SD) + P(NoPCOS|SD) = 0.203 + 0.104 = 0.307

Therefore, the probability of having skin darkening and having PCOS is P(PCOS|SD)/Normalizer = 0.203/0.307 = 0.661

The probability of having skin darkening and not having PCOS is P(NoPCOS|SD)/Normalizer = 0.104/0.307 = 0.339